hack/api-docs/build.sh

```diff
@@ -9,31 +9,43 @@
 set -euo pipefail
 
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+REPO_ROOT="${SCRIPT_DIR}/../.."
 DOCS_DIR="${SCRIPT_DIR}/../../docs"
-REFDOCS_PKG="github.com/elastic/gen-crd-api-reference-docs"
+REFDOCS_REPO="https://github.com/elastic/gen-crd-api-reference-docs.git"
 REFDOCS_VER="master"
-REFDOCS_BIN="$(go env GOPATH)/bin/$(basename $REFDOCS_PKG)"
 
-install_refdocs() {
+log() {
+    echo ">> $1"
+}
+
+build_docs() {
     local INSTALL_DIR=$(mktemp -d)
     (
+        log "Building refdoc binary..."
         cd $INSTALL_DIR
-        GO111MODULE=on go mod init github.com/elastic/eck-refdocs
-        GO111MODULE=on go get -u "${REFDOCS_PKG}@${REFDOCS_VER}"
+        git clone -q --depth 1 --branch $REFDOCS_VER --single-branch $REFDOCS_REPO
+        cd gen-crd-api-reference-docs
+        go mod edit --replace=github.com/elastic/cloud-on-k8s@latest=$REPO_ROOT
+        go build
     )
-}
 
-build_docs() {
+    local REFDOCS_BIN=${INSTALL_DIR}/gen-crd-api-reference-docs/gen-crd-api-reference-docs
     local TEMP_OUT_FILE=$(mktemp)
-    $REFDOCS_BIN -api-dir=github.com/elastic/cloud-on-k8s/pkg/apis \
-        -template-dir="${SCRIPT_DIR}/templates" \
-        -out-file=$TEMP_OUT_FILE \
-        -config="${SCRIPT_DIR}/config.json"
+    (
+        log "Generating API reference documentation..."
+        cd $REPO_ROOT
+        $REFDOCS_BIN -api-dir=./pkg/apis \
+            -template-dir="${SCRIPT_DIR}/templates" \
+            -out-file=$TEMP_OUT_FILE \
+            -config="${SCRIPT_DIR}/config.json" \
+            -alsologtostderr=false \
+            -logtostderr=false \
+            -stderrthreshold=ERROR
+        mv $TEMP_OUT_FILE "${DOCS_DIR}/api-docs.asciidoc"
+        log "API reference documentation generated successfully."
+    )
 
-    mv $TEMP_OUT_FILE "${DOCS_DIR}/api-docs.asciidoc"
+    rm -rf $INSTALL_DIR || log "Failed to clean-up $INSTALL_DIR"
```

Comment:
Each execution of the script creates 2 log files and create or update 2 symbolic links in `/tmp`. As there is already a cleaning effort, I wonder if this is voluntary to keep them?
```sh
> ll /tmp/g*
-rw-rw-r-- 1 thb thb 3,5K oct.  23 18:06 /tmp/gen-crd-api-reference-docs.fun.thb.log.INFO.20191023-180636.359
-rw-rw-r-- 1 thb thb 3,5K oct.  23 18:07 /tmp/gen-crd-api-reference-docs.fun.thb.log.INFO.20191023-180713.4436
-rw-rw-r-- 1 thb thb 2,9K oct.  23 18:06 /tmp/gen-crd-api-reference-docs.fun.thb.log.WARNING.20191023-180636.359
-rw-rw-r-- 1 thb thb 2,9K oct.  23 18:07 /tmp/gen-crd-api-reference-docs.fun.thb.log.WARNING.20191023-180713.4436
lrwxrwxrwx 1 thb thb   64 oct.  23 18:07 /tmp/gen-crd-api-reference-docs.INFO -> gen-crd-api-reference-docs.fun.thb.log.INFO.20191023-180713.4436
lrwxrwxrwx 1 thb thb   67 oct.  23 18:07 /tmp/gen-crd-api-reference-docs.WARNING -> gen-crd-api-reference-docs.fun.thb.log.WARNING.20191023-180713.4436
```